tests/microos/cockpit_check.pm

```diff
@@ -0,0 +1,51 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Basic check for cockpit service
+# Maintainer: qa-c team <qa-c@suse.de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use transactional;
+use utils qw(systemctl);
+
+sub run {
+    my ($self) = @_;
+
+    select_console 'root-console';
+
+    # Install cockpit if needed, this is needed for DVD flavor where
+    # Cockpit pattern is not selected during install
+    if (script_run('rpm -q cockpit') != 0) {
+        record_info('INFO', 'Installing Cockpit...');
+        trup_call 'pkg install cockpit';
+        check_reboot_changes;
+    }
+
+    record_info('Cockpit', script_output('rpm -qi cockpit'));
+
+    # Enable service
+    systemctl 'enable --now cockpit.socket';
+    systemctl 'start cockpit';
```

Comment:
Doesn't the service enable socket by default?

Reply:
I thought so, but not. See [this comment](https://bugzilla.suse.com/show_bug.cgi?id=1186737#c3)

Reply:
Not really.

```
localhost:~ # systemctl status cockpit
* cockpit.service - Cockpit Web Service
     Loaded: loaded (/usr/lib/systemd/system/cockpit.service; static)
     Active: inactive (dead)
TriggeredBy: * cockpit.socket
       Docs: man:cockpit-ws(8)
       
localhost:~ # systemctl enable --now  cockpit.socket
Created symlink /etc/systemd/system/sockets.target.wants/cockpit.socket -> /usr/lib/systemd/system/cockpit.socket.

localhost:~ # systemctl status cockpit
* cockpit.service - Cockpit Web Service
     Loaded: loaded (/usr/lib/systemd/system/cockpit.service; static)
     Active: inactive (dead)
TriggeredBy: * cockpit.socket
       Docs: man:cockpit-ws(8)
       
localhost:~ # systemctl start cockpit

localhost:~ # systemctl status cockpit
* cockpit.service - Cockpit Web Service
     Loaded: loaded (/usr/lib/systemd/system/cockpit.service; static)
     Active: active (running) since Thu 2021-06-03 10:19:46 UTC; 1s ago
TriggeredBy: * cockpit.socket
       Docs: man:cockpit-ws(8)
    Process: 2083 ExecStartPre=/usr/sbin/remotectl certificate --ensure --user=root --group=cockpit-wsinstance --selinux-type=etc_t (code=exited, status=0/SUCCESS)
   Main PID: 2087 (cockpit-ws)
      Tasks: 2
     CGroup: /system.slice/cockpit.service
             `-2087 /usr/lib/cockpit-ws
```

Reply:
Sorry, I meant vice versa. 

```
➜  ~ systemctl status cockpit
● cockpit.service - Cockpit Web Service
     Loaded: loaded (/usr/lib/systemd/system/cockpit.service; static)
     Active: inactive (dead)
TriggeredBy: ● cockpit.socket
       Docs: man:cockpit-ws(8)
➜  ~ systemctl start cockpit
➜  ~ systemctl status cockpit.service
● cockpit.service - Cockpit Web Service
     Loaded: loaded (/usr/lib/systemd/system/cockpit.service; static)
     Active: active (running) since Thu 2021-06-03 12:29:23 CEST; 4s ago
TriggeredBy: ● cockpit.socket
       Docs: man:cockpit-ws(8)
    Process: 9480 ExecStartPre=/usr/libexec/cockpit-certificate-ensure (code=exited, status=0/SUCCESS)
   Main PID: 9498 (cockpit-tls)
      Tasks: 1 (limit: 4915)
        CPU: 85ms
     CGroup: /system.slice/cockpit.service
             └─9498 /usr/libexec/cockpit-tls

Jun 03 12:29:23 eris systemd[1]: Starting Cockpit Web Service...
Jun 03 12:29:23 eris cockpit-certificate-ensure[9488]: /usr/libexec/cockpit-certificate-helper: line 32: sscg: command not found
Jun 03 12:29:23 eris cockpit-certificate-ensure[9489]: Generating a RSA private key
Jun 03 12:29:23 eris cockpit-certificate-ensure[9489]: .................................+++++
Jun 03 12:29:23 eris cockpit-certificate-ensure[9489]: ..+++++
Jun 03 12:29:23 eris cockpit-certificate-ensure[9489]: writing new private key to '0-self-signed.key'
Jun 03 12:29:23 eris cockpit-certificate-ensure[9489]: -----
Jun 03 12:29:23 eris systemd[1]: Started Cockpit Web Service.
➜  ~ systemctl status cockpit.socket
● cockpit.socket - Cockpit Web Service Socket
     Loaded: loaded (/usr/lib/systemd/system/cockpit.socket; disabled; vendor preset: disabled)
     Active: active (running) since Thu 2021-06-03 12:29:23 CEST; 8s ago
   Triggers: ● cockpit.service
       Docs: man:cockpit-ws(8)
     Listen: [::]:9090 (Stream)
    Process: 9470 ExecStartPost=/usr/share/cockpit/motd/update-motd  localhost (code=exited, status=0/SUCCESS)
    Process: 9477 ExecStartPost=/bin/ln -snf active.motd /run/cockpit/motd (code=exited, status=0/SUCCESS)
      Tasks: 0 (limit: 4915)
        CPU: 63ms
     CGroup: /system.slice/cockpit.socket

Jun 03 12:29:23 eris systemd[1]: Starting Cockpit Web Service Socket.
Jun 03 12:29:23 eris systemd[1]: Listening on Cockpit Web Service Socket.
```

Reply:
hm...or maybe just activating the socket is enough and later on it should trigger the service when it receives any activity. But yeah, neat picking :)

Reply:
Ah, ok just noticed the email notification with [details](https://bugzilla.suse.com/show_bug.cgi?id=1186737#c3).

Reply:
I have just tried and I will proceed this way. Let me update it.

Reply:
Please check latest change. The service is enabled automatically after curl command.

Reply:
> Please check latest change. The service is enabled automatically after curl command.

I guess it will fail after reboot as the service will be inactive. Other than that, looks good.